pkg/operation/botanist/backupentry/backupentry_test.go

```diff
@@ -0,0 +1,230 @@
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("BackupEntry", func() {
+	var (
+		ctrl *gomock.Controller
+
+		ctx              context.Context
+		c                client.Client
+		expected         *gardencorev1beta1.BackupEntry
+		values           *Values
+		log              logrus.FieldLogger
+		defaultDepWaiter component.DeployWaiter
+
+		mockNow *mocktime.MockNow
+		now     time.Time
+
+		name      = "be"
+		namespace = "namespace"
+		ownerRef  = metav1.NewControllerRef(&corev1.Namespace{}, corev1.SchemeGroupVersion.WithKind("Namespace"))
+
+		seedName            = "seed"
+		bucketName          = "bucket"
+		differentBucketName = "different-bucketname"
+		differentSeedName   = "different-seedname"
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+
+		mockNow = mocktime.NewMockNow(ctrl)
+
+		ctx = context.TODO()
+		log = logger.NewNopLogger()
+
+		s := runtime.NewScheme()
+		Expect(gardencorev1beta1.AddToScheme(s)).NotTo(HaveOccurred())
+
+		c = fake.NewFakeClientWithScheme(s)
+
+		values = &Values{
+			Name:           name,
+			Namespace:      namespace,
+			OwnerReference: ownerRef,
+			SeedName:       &seedName,
+			BucketName:     bucketName,
+		}
+
+		expected = &gardencorev1beta1.BackupEntry{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: gardencorev1beta1.SchemeGroupVersion.String(),
+				Kind:       "BackupEntry",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+				Annotations: map[string]string{
+					v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
+					v1beta1constants.GardenerTimestamp: now.UTC().String(),
+				},
+				Finalizers:      []string{"gardener"},
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Spec: gardencorev1beta1.BackupEntrySpec{
+				BucketName: bucketName,
+				SeedName:   &seedName,
+			},
+		}
+
+		defaultDepWaiter = New(log, c, values, time.Millisecond, 500*time.Millisecond)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#Deploy", func() {
+		BeforeEach(func() {
+			expected.ResourceVersion = "1"
+		})
+
+		Context("do not overwrite seed name", func() {
+			It("should create correct BackupEntry (newly created)", func() {
+				defer test.WithVars(&TimeNow, mockNow.Do)()
+				mockNow.EXPECT().Do().Return(now.UTC()).AnyTimes()
+
+				Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
+
+				actual := &gardencorev1beta1.BackupEntry{}
+				Expect(c.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, actual)).To(Succeed())
+
+				Expect(actual).To(DeepEqual(expected))
```

Comment:
Why we use a custom `DeepEqual` matcher? I replaced the with the `DeepEqual` usage with gomega `Equal` matcher (which uses `reflect.DeepEqual` under the hood) and it worked well. And I was wondering why we don't simply use `Equal`.

```suggestion
				Expect(actual).To(Equal(expected))
```

Reply:
The reason is to better show the mismatch/difference in case a test fails. While it's very hard to find it when using `Equal`, our custom `DeepEqual` matcher shows the YAMLs and makes it a lot easier to find the problem.